File: Data Structures & Algorithms/max-consecutive-ones/submission-2.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        current = 0
        _max = 0
        for n in nums:
            if n == 1:
                current += 1
                _max = max(current, _max)
            else:
                current = 0
        
        return _max

File: Data Structures & Algorithms/partition-equal-subset-sum/submission-2.rs
impl Solution {
    pub fn can_partition(nums: Vec<i32>) -> bool {
        let total: i32 = nums.iter().sum();

        if total % 2 != 0 {
            return false;
        }

        let target = total as usize / 2;
        let mut dp = vec![false; target + 1];
        dp[0] = true;

        for num in nums {
            let value = num as usize;
            if value > target {
                return false;
            }
            for cur_sum in (value..=target).rev() {
                dp[cur_sum] |= dp[cur_sum - value]
            }
        }

        dp[target]
    }
}
